src/components/PurchaseOrder/CloseOrder/CloseOrderModal.js

```diff
@@ -1,106 +1,108 @@
-  static propTypes = {
-    orderNumber: PropTypes.string,
-    closingReasons: PropTypes.arrayOf(PropTypes.object),
-    closeOrder: PropTypes.func.isRequired,
-    cancel: PropTypes.func.isRequired,
-  }
+  const onChangeReason = useCallback(
+    ({ target: { value } }) => (
+      setReason(value)
+    ),
+    [],
+  );
 
-  constructor(props) {
-    super(props);
+  const onChangeNote = useCallback(
+    ({ target: { value } }) => (
+      setNote(value)
+    ),
+    [],
+  );
 
-    this.state = {
-      reason: '',
-      note: '',
-    };
-  }
+  const onClose = useCallback(
+    () => closeOrder(reason, note),
+    [closeOrder, reason, note],
+  );
 
-  reasonChanged = (e) => {
-    this.setState({ reason: e.target.value });
-  }
-
-  noteChanged = (e) => {
-    this.setState({ note: e.target.value });
-  }
-
-  render() {
-    const { orderNumber, closingReasons, closeOrder, cancel } = this.props;
-    const reasons = getClosingReasonsOptions(closingReasons);
-
-    return (
-      <Modal
-        data-test-close-order-modal
-        label={<FormattedMessage id="ui-orders.closeOrderModal.title" values={{ orderNumber }} />}
-        open
+  const footer = (
+    <ModalFooter>
+      <Button
+        buttonStyle="primary"
+        data-test-close-order-modal-submit
+        disabled={!reason}
+        onClick={onClose}
       >
-        <Row>
-          <Col xs={12}>
-            <Select
-              autoFocus
-              label={<FormattedMessage id="ui-orders.closeOrderModal.reason" />}
-              data-test-closing-reasons
-              dataOptions={reasons}
-              onChange={this.reasonChanged}
-              placeholder=" "
-              defaultValue=""
-            >
-              {Object.keys(DEFAULT_CLOSE_ORDER_REASONS).map((key) => (
-                <FormattedMessage
-                  id={`ui-orders.closeOrderModal.closingReasons.${key}`}
-                  key={key}
-                >
-                  {(message) => <option value={DEFAULT_CLOSE_ORDER_REASONS[key]}>{message}</option>}
-                </FormattedMessage>
-              ))}
-            </Select>
-            <TextArea
-              label={<FormattedMessage id="ui-orders.closeOrderModal.notes" />}
-              onChange={this.noteChanged}
-            />
-          </Col>
-        </Row>
-        <Row>
-          <Col
-            className={css.buttonsLine}
-            xs={12}
+        <FormattedMessage id="ui-orders.closeOrderModal.submit" />
+      </Button>
+      <Button
+        data-test-close-order-modal-cancel
+        onClick={cancel}
+      >
+        <FormattedMessage id="ui-orders.closeOrderModal.cancel" />
+      </Button>
+    </ModalFooter>
+  );
+
+  return (
+    <Modal
+      data-test-close-order-modal
+      label={<FormattedMessage id="ui-orders.closeOrderModal.title" values={{ orderNumber }} />}
+      footer={footer}
+      open
+    >
+      <Row>
+        <Col xs={12}>
+          <Select
```

Comment:
Select is quite bad component for this, IMHO it could be improved a lot. Since there are could be tons of reasons added by user in settings it would be much nicer to have Selection with search input. What do you think?

Reply:
But there is an issue with dropdown using Selection in Modal. Need more time to investigate it.